update to zig 0.12

psyomn opened the pull request:
This should update the project to zig `0.12`.

I would like a little bit more time to test this against some of my personal scripts, but if you see nothing wrong with it, feel free to merge!

Cheers!

psyomn committed:
zig-0.12.0: update build.zig incantations

psyomn committed:
zig-0.12.0: fix const warnings

psyomn committed:
zig-0.12.0: replace std.mem.copy with @memcpy builtin

ikskuh commented on build.zig:
```diff
@@ -6,9 +6,9 @@ const example_list = [_][]const u8{
-pub fn build(b: *std.build.Builder) void {
+pub fn build(b: *std.Build) void {
     const gemtext = b.createModule(.{
-        .source_file = .{ .path = "src/gemtext.zig" },
+        .root_source_file = .{ .path = "src/gemtext.zig" },
     });
```
I think `addModule("gemtext", …)` is better here, we should also expose the module itself to the public

psyomn replied:
Yes that totally makes sense, thank you!

psyomn committed:
Expose gemtext module through addModule

This makes it so that others can include the gemtext module through
Zig's package manager.
